client/my-sites/checkout/upsell-nudge/index.tsx

```diff
@@ -596,12 +557,11 @@ export default connect(
 				: undefined;
 
 		return {
-			isFetchingStoredCards: areStoredCardsLoading,
 			cards,
 			currencyCode: getCurrentUserCurrencyCode( state ),
 			currentPlanTerm,
 			isLoading:
-				isFetchingCards ||
+				areStoredCardsLoading ||
```

Comment:
```suggestion
				! hasLoadedCardsFromServer ||
```

Reply:
I've looked more into the variables `isFetchingCards` and `hasLoadedCardsFromServer`. It appears that:
* `isFetchingCards` is set to `true` when cards are being fetched, it's set to `false` before the fetch starts, and after the fetch is completed
* `hasLoadedCardsFromServer` is set to `true` only when the cards are loaded

In terms of setting the `isLoading` set, all we'd really need is just to check for `! hasLoadedCardsFromServer`. It looks like we can get rid of `areStoredCardsLoading` and `isFetchingCards` altogether :) 

I've tested all three upsells with the change, and all upsell components rendered as expected.

* Professional Email

https://user-images.githubusercontent.com/104910361/191877703-d213c3b9-4a96-45cb-a481-87a217d4a714.mov

* Plan Upgrade

https://user-images.githubusercontent.com/104910361/191877690-072cce50-8d23-422c-9c3b-47281303a406.mov

* Quick Start Session

https://user-images.githubusercontent.com/104910361/191877659-e9ee1594-fd24-4f22-a355-0ad49498ca36.mov

Reply:
I thought about it too, but in this case, it won't be set to `true` during lifecycle of the component, therefore - we totally change the behaviour, so potentially we can break something [here](https://github.com/Automattic/wp-calypso/blob/add/post-checkout-professional-email-tracking-event/client/my-sites/checkout/upsell-nudge/index.tsx#L131).

And just for the records, we have to rename `isLoading` to `isLoaded`.

Reply:
Got it, thanks for walking me through the logic! No let's not do the refactoring and focus on the initial feature, which is to add tracking :)